airflow/providers/amazon/aws/sensors/rds.py

```diff
@@ -38,10 +38,10 @@ def __init__(self, *args, aws_conn_id: str = "aws_conn_id", hook_params: Optiona
         hook_params = hook_params or {}
         self.hook = RdsHook(aws_conn_id=aws_conn_id, **hook_params)
         self.target_statuses: List[str] = []
+        self.check_status_field = "Status"
```

Comment:
Current sensors rely on the same `botocore` response structure. The response to [`describe_db_instances`](https://botocore.amazonaws.com/v1/documentation/api/latest/reference/services/rds.html#RDS.Client.describe_db_instances) uses a different field name to describe the state of the requested resource. I chose to define the field name as an instance variable because..
- no need to update other sensor classes
- no need to modify method signatures
- minimal code changes
- it makes sense to define "sensor specific metadata" on the sensor class/instance

Reply:
I think this variable could be in `_check_item` method.
Based on `item_type` you could identify the status field

Reply:
You're right, but that requires adding an `if-else` conditional statement. Is the current approach OK with you?

Reply:
I think it is OK to add one more `if-else`.
Actually, I meant something like:
```
    def _check_item(self, item_type: str, item_name: str) -> bool:
        """Get certain item from `_describe_item()` and check its status"""
        if item_type == 'db_instance':
                status_field = 'DBInstanceStatus'
        else:
                status_field = 'Status'

        try:
            items = self._describe_item(item_type, item_name)
        except ClientError:
            return False
        else:
            return bool(items) and any(
                map(lambda status: items[0][status_field].lower() == status, self.target_statuses)
            )
```
WDYT?

Reply:
By adding if/else logic to the base class, we're making it "aware" of the details of the subclass. I think a better approach is to let the subclass tell us what it needs.

I checked the `botocore` documentation and it looks like "DBInstanceStatus" and "Status" are the only 2 options, so I'll go ahead and make the change to finish up this PR.

Reply:
Changed in https://github.com/apache/airflow/pull/26003/commits/afefe365706c625bc0deebbf9bc51362ce2bf367